README.md
This is the GIT repository for Wally-Core

CHANGELOG.txt
Drupal 6.26, 2012-05-02 -222
----------------------
- Fixed a small number of bugs.
- Made code documentation improvements.

Drupal 6.25, 2012-02-29
----------------------
- Fixed regressions introduced in Drupal 6.24 only.

Drupal 6.24, 2012-02-01
----------------------
- Improved performance of search indexing and user operations by adding indexes.
- Fixed issues with themes getting disabled due to missing locking in
  system_theme_data().
- Fix issue with blocks being disabled on updates in _block_rehash().
- Further improvements to PHP 5.3, PHP 4 and PostgreSQL compatibility.
- Improved code documentation at various places.
- Fixed a variety of other bugs.

Drupal 6.23, 2012-02-01
----------------------
- Fixed security issues (Cross site scripting), see SA-CORE-2012-001.

Drupal 6.22, 2011-05-25
----------------------
- Made Drupal 6 work better with IIS and Internet Explorer.
- Fixed .po file imports to work better with custom textgroups.
- Improved code documentation at various places.
- Fixed a variety of other bugs.

Drupal 6.21, 2011-05-25
----------------------
- Fixed security issues (Cross site scripting), see SA-CORE-2011-001.

Drupal 6.20, 2010-12-15
----------------------
- Fixed a variety of small bugs, improved code documentation.

Drupal 6.19, 2010-08-11
----------------------
- Fixed a variety of small bugs, improved code documentation.

Drupal 6.18, 2010-08-11
----------------------
- Fixed security issues (OpenID authentication bypass, File download access
  bypass, Comment unpublishing bypass, Actions cross site scripting),
  see SA-CORE-2010-002.

Drupal 6.17, 2010-06-02
----------------------
- Improved PostgreSQL compatibility
- Better PHP 5.3 and PHP 4 compatibility
- Better browser compatibility of CSS and JS aggregation
- Improved logging for login failures
- Fixed an incompatibility with some contributed modules and the locking system
- Fixed a variety of other bugs.

Drupal 6.16, 2010-03-03
----------------------
- Fixed security issues (Installation cross site scripting, Open redirection,
  Locale module cross site scripting, Blocked user session regeneration),
  see SA-CORE-2010-001.
- Better support for updated jQuery versions.
- Reduced resource usage of update.module.
- Fixed several issues relating to support of install profiles and
  distributions.
- Added a locking framework to avoid data corruption on long operations.
- Fixed a variety of other bugs.

Drupal 6.15, 2009-12-16
----------------------
- Fixed security issues (Cross site scripting), see SA-CORE-2009-009.
- Fixed a variety of other bugs.

Drupal 6.14, 2009-09-16
----------------------
- Fixed security issues (OpenID association cross site request forgeries,
  OpenID impersonation and File upload), see SA-CORE-2009-008.
- Changed the system modules page to not run all cache rebuilds; use the
  button on the performance settings page to achieve the same effect.
- Added support for PHP 5.3.0 out of the box.
- Fixed a variety of small bugs.

Drupal 6.13, 2009-07-01
----------------------
- Fixed security issues (Cross site scripting, Input format access bypass and
  Password leakage in URL), see SA-CORE-2009-007.
- Fixed a variety of small bugs.

Drupal 6.12, 2009-05-13
----------------------
- Fixed security issues (Cross site scripting), see SA-CORE-2009-006.
- Fixed a variety of small bugs.

Drupal 6.11, 2009-04-29
----------------------
- Fixed security issues (Cross site scripting and limited information
  disclosure), see SA-CORE-2009-005
- Fixed performance issues with the menu router cache, the update
  status cache and improved cache invalidation
- Fixed a variety of small bugs.

Drupal 6.10, 2009-02-25
----------------------
- Fixed a security issue, (Local file inclusion on Windows),
  see SA-CORE-2009-003
- Fixed node_feed() so custom fields can show up in RSS feeds.
- Improved PostgreSQL compatibility.
- Fixed a variety of small bugs.

Drupal 6.9, 2009-01-14
----------------------
- Fixed security issues, (Access Bypass, Validation Bypass and Hardening
  against SQL injection), see SA-CORE-2009-001
- Made HTTP request checking more robust and informative.
- Fixed HTTP_HOST checking to work again with HTTP 1.0 clients and
  basic shell scripts.
- Removed t() calls from all schema documentation. Suggested best practice
  changed for contributed modules, see http://drupal.org/node/322731.
- Fixed a variety of small bugs.

Drupal 6.8, 2008-12-11
----------------------
- Removed a previous change incompatible with PHP 5.1.x and lower.

Drupal 6.7, 2008-12-10
----------------------
- Fixed security issues, (Cross site request forgery and Cross site scripting), see SA-2008-073
- Updated robots.txt and .htaccess to match current file use.
- Fixed a variety of small bugs.

Drupal 6.6, 2008-10-22
----------------------
- Fixed security issues, (File inclusion, Cross site scripting), see SA-2008-067
- Fixed a variety of small bugs.

Drupal 6.5, 2008-10-08
----------------------
- Fixed security issues, (File upload access bypass, Access rules bypass,
  BlogAPI access bypass), see SA-2008-060.
- Fixed a variety of small bugs.

Drupal 6.4, 2008-08-13
----------------------
- Fixed a security issue (Cross site scripting, Arbitrary file uploads via
  BlogAPI, Cross site request forgeries and Various Upload module
  vulnerabilities), see SA-2008-047.
- Improved error messages during installation.
- Fixed a bug that prevented AHAH handlers to be attached to radios widgets.
- Fixed a variety of small bugs.

Drupal 6.3, 2008-07-09
----------------------
- Fixed security issues, (Cross site scripting, cross site request forgery,
  session fixation and SQL injection), see SA-2008-044.
- Slightly modified installation process to prevent file ownership issues on
  shared hosts.
- Improved PostgreSQL compatibility (rewritten queries; custom blocks).
- Upgraded to jQuery 1.2.6.
- Performance improvements to search, menu handling and form API caches.

metriweb/mwTag.js
/* MetriWeb Javascript file */
/* Version 5.1 -- release 1.0 -- 2010-01-06 */

/* Please, do NEVER modify this monitored file! */

var mw=new Object({
V:"5.1",
R:"1.0",
i:new Image(),
p:0,
l:0,
D:".metriweb.be",
belgian:1,
r:function(u1,u2){
	if(mw.i.width){
		if(mw.i.width>1){
			var nS=document.createElement("script");
			nS.setAttribute("src", u1+"td"+u2+"&w="+mw.i.width+'&T='+new Date().getTime());
			document.getElementsByTagName("head").item(0).appendChild(nS);
		}
	}
	else
		setTimeout("mw.r('"+u1.replace(/'/,"\\'")+"','"+u2.replace(/'/,"\\'")+"')",1000);
},
W:function(u1,u2){
	if(mw.l){
		if(mw.l++>1){
			var nS=document.createElement("script");
			nS.setAttribute("src", u1+"sd"+u2+'&T='+new Date().getTime());
			document.getElementsByTagName("head").item(0).appendChild(nS);
		}else mw.r(u1,u2)
	}else{
		mw.i.src=u1+"dyn"+u2;
		if(!mw.p){
			if(window.addEventListener)window.addEventListener('load',function(){mw.l++;mw.W(u1,u2)},false);
			else if(window.attachEvent)window.attachEvent('onload',function(){mw.l++;mw.W(u1,u2)});
		}
	}
},
n:function(d){
	var k="0000";
	var t=d.split(".");
	if(parseInt(k+t[t.length-1]))mw.d=d;
	else mw.d="."+t[t.length-2]+"."+t[t.length-1];
	t="charCodeAt(0)";
	var c=(k+eval("\""+mw.d.split("").join("\"."+t+"+\"")+"\"."+t));
	t=k+k+k+k+new Date().getTime();
	t=t.substr(t.length-16,16);
	return t.substr(0,13)+"."+t.substr(13,3)+c.substr(c.length-3,3)+".5";
},
g:function(s,f){
	var c;
	var o=document.cookie.indexOf(s+"=");
	if(o<0){
		c=f(document.domain);
		document.cookie=s+"="+c+"; domain="+mw.d+"; path=/; expires="+
			new Date(new Date().getTime()+365*24*3600000).toGMTString()+";";
	}else{
		var e=document.cookie.indexOf(";",o);
		if(e<0)e=document.cookie.length;
		c=document.cookie.substring(o+4,e);
	}
	return c;
},
H:function(){
	var e=document.getElementById("metriwebLayer");
	if (e)
		e.style.visibility="hidden";
},
e:window.screen?window.screen.width+"x"+window.screen.height+"/"+window.screen.colorDepth:"-",
m:"0x0",
t:undefined,
k:undefined,
x:undefined,
s:undefined,
PR:0,
PV:0,
to:0,
pr:document.referrer?escape(document.referrer):"-",
RIA:function(action,keyword,origin,into){
	mw.W('http://'+mw.t+mw.D+'/','/'+mw.t+'/mw.cgi?page='+keyword+
		(mw.x?('&q='+mw.x):'')+(mw.s?('&s='+mw.s):'')+
		'&c='+mw.c+'&v='+mw.V+"."+mw.R+'&p='+mw.p+'&d='+Math.round(new Date().getTime())+
		'&e='+mw.e+'&a='+action+'&r='+mw.pr+'&f='+mw.f+
		'&m='+mw.m+'&o='+origin+'&i='+into+'&R='+Math.random());
},
em:function(e) {
	if(!e)var e=window.event;
	mw.m=e.screenX+"x"+e.screenY;
},
eb:function() {
	mw.PV=new Date().getTime();
	if(mw.to){clearTimeout(mw.to);mw.to=0};
},
ef:function() {
	if(!mw.PV){
		if(new Date().getTime()-mw.PR>15e3){
			if(mw.to){clearTimeout(mw.to);mw.to=setTimeout("mw.to=0;mw.RIA('page.idle','"+mw.k+"','','')",3e4);};
			mw.RIA("page.view",mw.k,"","");
		}
		mw.PV=new Date().getTime();
	}
}
});
mw.c=mw.g("mwc",mw.n);
if(window.addEventListener){
	window.addEventListener('focus',mw.ef,false);
	window.addEventListener('blur',mw.eb,false);
	window.addEventListener('mousemove',mw.em,false);
}else if(window.attachEvent){
	window.attachEvent('onfocus',mw.ef);
	window.attachEvent('onblur',mw.eb);
	window.attachEvent('onmousemove',mw.em);
}
function metriwebTag (tag,keyword,extra,refresh){
	if(mw.belgian&&!mw.p){
		try{
			mw.t=tag.replace(/[^a-zA-Z0-9\-/]/g,"").substring(0,24);
			mw.k=keyword.replace(/[^a-zA-Z0-9_/]/g,"").substring(0,24);
			if(extra.match(/en|fr|nl|ge/)){mw.x=extra}else mw.x="na";
			mw.s=refresh?parseInt(refresh):undefined;
			mw.PR=new Date().getTime();
			mw.to=setTimeout("mw.to=0;mw.RIA('page.idle','"+mw.k+"','','')",3e4);
			mw.f=(mw.t==tag?0:1)+(mw.k==keyword?0:2)+(mw.x==extra?0:4)+(mw.s==refresh?0:8);
			mw.RIA("page.request",mw.k,"","");
			mw.p++;
		}catch (e){}
	}
}
function metriwebRIA (action,keyword,origin)
{
	if(mw.belgian&&mw.PR){
		try{
			a=action.replace(/[^a-zA-Z0-9\./]/g,"").substring(0,24);
			k=keyword?keyword.replace(/[^a-zA-Z0-9_/]/g,"").substring(0,24):mw.k;
			o=origin?origin.replace(/[^a-zA-Z0-9\-/]/g,"").substring(0,24):"";
			if(mw.to){clearTimeout(mw.to);mw.to=0};
			mw.RIA(a,k,o,mw.k);
			mw.p++;
		}catch (e){}
	}
}

/* (c) 2000-2010, DouWère/MetriWare */
